core/src/Cabana_Slice.hpp

```diff
@@ -22,6 +22,333 @@
 #include <cstdlib>
 #include <type_traits>
 
+//---------------------------------------------------------------------------//
+namespace Kokkos
+{
+
+//---------------------------------------------------------------------------//
+// Cabana Slice Layout
+struct LayoutCabanaSlice : public LayoutStride
+{
+    typedef LayoutCabanaSlice array_layout;
+
+    LayoutCabanaSlice( LayoutCabanaSlice const & ) = default ;
+    LayoutCabanaSlice( LayoutCabanaSlice && ) = default ;
+    LayoutCabanaSlice & operator = ( LayoutCabanaSlice const & ) = default ;
+    LayoutCabanaSlice & operator = ( LayoutCabanaSlice && ) = default ;
+
+    KOKKOS_INLINE_FUNCTION
+    explicit constexpr
+    LayoutCabanaSlice( size_t N0 = 0, size_t S0 = 0,
+                       size_t N1 = 0, size_t S1 = 0,
+                       size_t N2 = 0, size_t S2 = 0,
+                       size_t N3 = 0, size_t S3 = 0,
+                       size_t N4 = 0, size_t S4 = 0,
+                       size_t N5 = 0, size_t S5 = 0,
+                       size_t N6 = 0, size_t S6 = 0,
+                       size_t N7 = 0, size_t S7 = 0 )
+    : LayoutStride( N0, S0,
+                    N1, S1,
+                    N2, S2,
+                    N3, S3,
+                    N4, S4,
+                    N5, S5,
+                    N6, S6,
+                    N7, S7 )
```

Comment:
I believe you should be able to use variadic templates and just forward all arguments to the constructor of `Kokkos::LayoutStride`.

Reply:
Actually, just remove that constructor definition and replace it with `using LayoutStride::LayoutStride;`